packages/amplify-e2e-tests/src/rds-v2-tests-common/rds-field-auth-apikey.ts

```diff
@@ -0,0 +1,619 @@
+    const setupDatabase = async (): Promise<void> => {
+      const dbConfig = {
+        identifier,
+        engine,
+        dbname: database,
+        username,
+        password,
+        region,
+      };
+
+      const db = await setupRDSInstanceAndData(dbConfig, queries);
+      port = db.port;
+      host = db.endpoint;
+    };
+
+    const cleanupDatabase = async (): Promise<void> => {
+      await deleteDBInstance(identifier, region);
+    };
+
+    const initProjectAndImportSchema = async (): Promise<void> => {
+      await initJSProjectWithProfile(projRoot, {
+        disableAmplifyAppCreation: false,
+        name: projName,
+      });
+
+      const metaAfterInit = getProjectMeta(projRoot);
+      region = metaAfterInit.providers.awscloudformation.Region;
+      await setupDatabase();
+
+      const rdsSchemaFilePath = path.join(projRoot, 'amplify', 'backend', 'api', apiName, 'schema.sql.graphql');
+
+      await addApiWithAllAuthModes(projRoot, { transformerVersion: 2, apiName });
+
+      await importRDSDatabase(projRoot, {
+        database,
+        engine,
+        host,
+        port,
+        username,
+        password,
+        useVpc: true,
+        apiExists: true,
+      });
+
+      const schema = /* GraphQL */ `
+        input AMPLIFY {
+          engine: String = "${engineName}"
+        }
+
+        # Api Key can't delete a record
+        # Api key can't update SSN field
+        # Api Key can't read SSN field 
+        # Api Key can create a record with all fields, but can't read SSN response
+        # Api Key subscription can't read SSN field
+        type Person1 @model @auth(rules: [{allow: public, provider: apiKey}, {allow: public, provider: iam}]) {
+          id: Int! @primaryKey
+          firstName: String
+          lastName: String
+          ssn: String @auth(rules: [{allow: public, provider: iam}, {allow: public, provider: apiKey, operations: [create]}])
+        }
+
+        # Api Key can't create a record with SSN value
+        # Api Key can create a record without a SSN value
+        # Api Key can't update a record with SSN value
+        # Api Key can't delete a record
+        type Person2 @model @auth(rules: [{allow: public, provider: apiKey}, {allow: public, provider: iam}]) {
+          id: Int! @primaryKey
+          firstName: String
+          lastName: String
+          ssn: String @auth(rules: [{allow: public, provider: iam}, {allow: public, provider: apiKey, operations: [read]}])
+        }
+
+        # Api Key can't create a record with SSN value
+        # Api Key can delete a record
+        type Person3 @model @auth(rules: [{allow: public, provider: apiKey}, {allow: public, provider: iam}]) {
+          id: Int! @primaryKey
+          firstName: String
+          lastName: String
+          ssn: String @auth(rules: [{allow: public, provider: iam}, {allow: public, provider: apiKey, operations: [delete]}])
+        }
+      `;
+      writeFileSync(rdsSchemaFilePath, schema, 'utf8');
```

Comment:
missing test for update operation using custom function provider. Can be followed up.